java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java

```diff
@@ -1,34 +1,158 @@
 //
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
-//
-// Ice version 3.7.10
-//
-// <auto-generated>
-//
-// Generated from file `Properties.ice'
-//
-// Warning: do not edit this file.
-//
-// </auto-generated>
-//
 
 package com.zeroc.Ice;
 
+import com.zeroc.IceInternal.Property;
+
 /**
  * A property set used to configure Ice and Ice applications. Properties are key/value pairs, with
  * both keys and values being strings. By convention, property keys should have the form
  * <em>application-name</em>[.<em>category</em>[.<em>sub-category</em>]].<em>name</em>.
  */
-public interface Properties {
+public final class Properties {
+  static class PropertyValue {
+    public PropertyValue(PropertyValue v) {
```

Comment:
Do we need to keep this constructor, or just `clone` like in C#?

Reply:
I meant to remove it and forgot!